W-15287123 extract pages from pdf (#354)
mulesoft/docs-rpa: rpa-builder/6.6/modules/ROOT/pages/toolbox-file-operations-extract-pages-from-pdf-file.adoc

= Extract Pages from PDF File

Extract pages from a PDF file, customize the output, and create a new PDF file with the configured settings.

== Properties

* *File Settings*
* *Create Target PDF if it doesn't exist*
+ 
Select this option to create the PDF file if the specified output file name doesn't exist. Otherwise, the action step fails if the file doesn't exist. 
* *Extraction Page Range* 
+

> The pages to extract from the input PDF. Use commas (`,`) to separate multiple ranges and `EOF` to specific the end of file. For example: `3-4, 8, 11-EOF`.

Reviewer: I think it's worth mentioning here that you can also put in reverse order ranges. For example, a range of 5-3 will copy the pages 5, 4, 3 in that order into the new document. This also works with EOF, so if you wanted to reverse an entire PDF file, you can put in EOF-1.

Author: Fixed

> The pages to extract from the input PDF. Use commas (`,`) to separate multiple ranges and `EOF` to specific the end of file. For example: `3-4, 8, 11-EOF`.

Author: ```suggestion
The pages to extract from the input PDF. Use commas (`,`) to separate multiple ranges and `EOF` to specify the end of the file. This action step reads the pages in the specified order. For example, the value`1-5, EOF-6` outputs a file that contains pages from 1 to 5 in the original order and the remaining pages in reverse order, from last to 6.

```

= Extract Pages from PDF File

Extract pages from a PDF file, customize the output, and create a new PDF file with the configured settings.

== Properties

> * *File Settings*

Reviewer 2: No description under this one?

Author: This is not a property, it's the UI container that groups the next properties.

Reviewer 2: Should the nesting reflect that?

Author: Yes, you are right. Fixed it.

* *Create Target PDF if it doesn't exist*
+ 
Select this option to create the PDF file if the specified output file name doesn't exist. Otherwise, the action step fails if the file doesn't exist. 
* *Extraction Page Range* 
+

> The pages to extract from the input PDF. Use commas (`,`) to separate multiple ranges and `EOF` to specify the end of the file. This action step reads the pages in the specified order. For example, the value`1-5, EOF-6` outputs a file that contains pages from 1 to 5 in the original order and the remaining pages in reverse order, from last to 6.

Reviewer 2: ```suggestion
The pages to extract from the input PDF. Use commas (`,`) to separate multiple ranges and `EOF` to specify the end of the file. This action step reads the pages in the specified order. For example, the value `1-5, EOF-6` outputs a file that contains pages from one to five in the original order and the remaining pages in reverse order, from last to six.

```

* *Input File Directory Path* 
+

> The path to the directory where the input file is. 

Reviewer 2: ```suggestion
Specify the path to the directory where the input file is. 

```

Reviewer 2: (Any way to make these descriptions parallel? All imperative or all noun phrases?)

* *Input File is Password Protected* 
+

> Select this option if the input file is password protected and specify the password in the Wizard. 

Reviewer 2: Is Wizard a proper noun here?

Reviewer 2: If the input file is password protected, select this option and specify the password in the wizard.

* *Input File Name* 
+

> The name of the input file. 

Reviewer 2: I think Diane prefers leaving off the article for incomplete sentences (noun descriptions):

Name of the input file.

* *Insert Mode* 
+
Select where to insert the extracted content in the output file. Available options: `StartOfDocument`, `SpecificPage`, or `EndOfDocument`. 
* *Insert Position* 
+
Select the page after which to insert the extracted content. If the PDF file doesn't exist, and *Create Target PDF if it doesn't exist* is selected, the action step inserts the pages at the beginning of the document. 
* *Output File Directory Path* 
+

> The path to the directory where the output file is. 

Reviewer 2: Path to the output file directory?

* *Output File is Password Protected* 
+

> Select this option if the output file is password protected and specify the password in the Wizard. 

Reviewer 2: See suggestion on line 20.

* *Output File Name* 
+

> The name of the output file. 

Reviewer 2: article

== Inbound Variables

* *Input File Directory Path* 
* *Input File Name* 

> * *InputFilePassword* 

Reviewer 2: Why is this CamelCase and the others are not?

Author: Because that's how it shows in the product

Reviewer 2: Weird! Any way to get them to change it to be consistent?

* *Extraction Page Range* 
* *Output File Directory Path* 
* *Output File Name* 

> * *OutputFilePassword* 

Reviewer 2: CamelCase correct?

* *Insert Position*

== Outbound Variables

> The _Extract Pages from PDF File_ action step has no outbound variables.

Reviewer 2: Should this be bold? Is it a UI element?

Reviewer 2: I guess action steps are italic?

= Extract Pages from PDF File

> Extract pages from a PDF file, customize the output, and create a new PDF file with the configured settings.

Author: ```suggestion
Extract pages from a PDF file, customize the output, and create a new PDF file with the configured settings. Use this action step to split PDF files, combine multiple PDF files into one, or extract specific pages to create a new PDF file.

```
